### PR TITLE
feat: http query deduplication api

### LINKED ISSUE #11710: Feature:  support deduplication on stage attachment api
**Summary**
To ensure data ingestion idempotency, databend has already support to deduplicate DML through deduplication label
https://databend.rs/doc/sql-commands/setting-cmds/set-var

Here for cross-language driver integration, we could add a rest api field for the label

### PR DESCRIPTION
**Pull request by akoshchiy**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

Added http query deduplication via `X-DATABEND-QUERY-ID` header.

Closes #11710

### CODE REVIEW

**Comment by sundy-li** [comment]
Hello, we have decoupled `query_id` and `deduplicate_label`.

The `query_id` is always generated inside the server.

So how about introducing a header named `X-DATABEND-DEDUPLICATED_LABEL` to inject the label settings.

File: src/query/service/src/servers/http/middleware.rs [hunk 1]
```diff
@@ -41,6 +41,9 @@ use crate::auth::Credential;
 use crate::servers::HttpHandlerKind;
 use crate::sessions::SessionManager;
 use crate::sessions::SessionType;
+
+const DEDUPLICATE_LABEL: &str = "X-DATABEND-DEDUPLICATE_LABEL";
```
Comment (sundy-li) [inline src/query/service/src/servers/http/middleware.rs:45]: I made a mistake, should be `"X-DATABEND-DEDUPLICATE-LABEL"`

File: tests/suites/1_stateful/07_stage_attachment/07_0002_insert_with_stage_deduplicate.sh [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+EOF
+
+aws --endpoint-url ${STORAGE_S3_ENDPOINT_URL} s3 cp s3://testbucket/admin/data/sample.csv s3://testbucket/admin/stage/internal/s1/sample.csv >/dev/null
+
+## Copy from internal stage
+echo "CREATE STAGE s1 FILE_FORMAT = (TYPE = CSV)" | $MYSQL_CLIENT_CONNECT
+echo "list @s1" | $MYSQL_CLIENT_CONNECT | awk '{print $1}'
+
+## Insert with stage use http API
+curl -s -u root: -XPOST "http://localhost:${QUERY_HTTP_HANDLER_PORT}/v1/query" --header 'Content-Type: application/json' --header 'X-DATABEND-DEDUPLICATE_LABEL: insert1' -d '{"sql": "insert into sample (Id, City, Score) values", "stage_attachment": {"location": "@s1/sample.csv"}}' | jq -r '.stats.scan_progress.bytes, .stats.write_progress.bytes, .error'
```
Comment (sundy-li) [inline tests/suites/1_stateful/07_stage_attachment/07_0002_insert_with_stage_deduplicate.sh:27]: X-DATABEND-DEDUPLICATE-LABEL

File: tests/suites/1_stateful/07_stage_attachment/07_0002_insert_with_stage_deduplicate.sh [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+echo "select * from sample" | $MYSQL_CLIENT_CONNECT
+
+## Insert again with the same deduplicate_label will have no effect
+curl -s -u root: -XPOST "http://localhost:${QUERY_HTTP_HANDLER_PORT}/v1/query" --header 'Content-Type: application/json' --header 'X-DATABEND-DEDUPLICATE_LABEL: insert1' -d '{"sql": "insert into sample (Id, City, Score) values", "stage_attachment": {"location": "@s1/sample.csv"}}' | jq -r '.stats.scan_progress.bytes, .stats.write_progress.bytes, .error'
```
Comment (sundy-li) [inline tests/suites/1_stateful/07_stage_attachment/07_0002_insert_with_stage_deduplicate.sh:32]: .

**Review by sundy-li — APPROVED** [review]

**Comment by BohuTANG** [comment]
Thanks for you contribution.

**Pull request merged**